### DISCUSSION TITLE
db.rawQuery on endpoint got `Converting object to an encodable object failed: Instance of 'DateTime'` error

### DISCUSSION THREAD

**Discussion by y3l1n4ung** [serverpod/serverpod#1093] [Q&A] [answered]
**Describe the bug**
I am encountering an error when using db.rawQuery in my code. The error message says "Converting object to an encodable object failed: Instance of 'DateTime'".

**To Reproduce**
1.after I call this endpoint 

```
Future<String?> fetchBlogWithAuthor(Session session, Author author) async {
    try {
      final x = await session.db.query('''SELECT b.*,
       ROW_TO_JSON(a) AS "authorJson"
FROM blog_tbl b
JOIN author_tbl a ON b."authorId" = a.id;''');
      print(x);

      print(x.first.runtimeType);

      return jsonEncode(x.first);
    } catch (e) {
      session.log(
        'Oops, something went wrong',
        level: LogLevel.warning,
        exception: e,
        // stackTrace: ,
      );
      return null;
    }
  }

```

2. I got this error ```Converting object to an encodable object failed: Instance of 'DateTime'```

**Expected behavior**
I just want to return like that
```
 [
	{
		"id": 8,
		"title": "testing",
		"content": "testing 123",
		"author": "yelinaung",
		"createAt": "2023-06-26 17:16:35.467889",
		"updateAt": "2023-06-26 17:16:35.467889",
		"authorId": 1,
		"authorJson": {
			"id": 1,
			"name": "YeLinAung",
			"createAt": "2023-06-26T17:15:16.833233",
			"updateAt": "2023-06-26T17:15:16.833233"
		}
	}
]
```

**Screenshots**
N/A

**Serverpod versions**
 -  _CLI Version:  1.1.0.
 - _Library version: 1.1.0

**Platform information**
 "macos_arm64"

**Additional context**

**Comment by vlidholt** [comment, marked as answer]
This isn't really an issue with Serverpod, but how you are using jsonEncode. jsonEncode doesn't have support for DateTime out of the box, but you will need a handler that serialize the objects the way you want them. There is an example in the Dart documentation.

```dart
final jsonText = jsonEncode({'cc': cc},
      toEncodable: (Object? value) => value is CustomClass
          ? CustomClass.toJson(value)
          : throw UnsupportedError('Cannot convert to JSON: $value'));
```

Read more here: https://api.flutter.dev/flutter/dart-convert/jsonEncode.html
